components/brave_wallet/browser/meld_integration_service.cc

```diff
@@ -128,6 +132,193 @@ bool NeedsToParseResponse(const int http_error_code) {
+
+  if (session_data->lock_fields && session_data->lock_fields->size()) {
+    base::Value::List lfs;
+    for (const auto& fld : *session_data->lock_fields) {
+      lfs.Append(fld);
+    }
+    cbsd.Set("lockFields", std::move(lfs));
+  }
+
+  if (session_data->payment_method_type) {
+    cbsd.Set("paymentMethodType", session_data->payment_method_type.value());
+  }
+
+  if (session_data->redirect_url) {
+    cbsd.Set("redirectUrl", session_data->redirect_url.value());
+  }
+
+  cbsd.Set("serviceProvider", session_data->service_provider);
+  cbsd.Set("sourceAmount", session_data->source_amount);
+  cbsd.Set("sourceCurrencyCode", session_data->source_currency_code);
+  if (session_data->wallet_address) {
+    cbsd.Set("walletAddress", session_data->wallet_address.value());
+  }
+
+  if (session_data->wallet_tag) {
+    cbsd.Set("walletTag", session_data->wallet_tag.value());
+  }
+
+  cbwr.Set("sessionData", std::move(cbsd));
+  cbwr.Set("sessionType", "SELL");
+
+  FillCustomerData(customer_data, cbwr);
+
+  return cbwr;
+}
+
+std::optional<base::Value::Dict> GetCryptoTransferWidgetPayload(
+    brave_wallet::mojom::CryptoTransferSessionDataPtr session_data,
+    brave_wallet::mojom::CryptoWidgetCustomerDataPtr customer_data) {
+  if (!session_data) {
+    return std::nullopt;
+  }
+
+  base::Value::Dict cbwr;
+  base::Value::Dict cbsd;
+
+  if (session_data->country_code) {
+    cbsd.Set("countryCode", session_data->country_code.value());
+  }
+
+  if (session_data->institution_id) {
+    cbsd.Set("institutionId", session_data->institution_id.value());
+  }
+
+  if (session_data->lock_fields && session_data->lock_fields->size()) {
+    base::Value::List lfs;
+    for (const auto& fld : *session_data->lock_fields) {
+      lfs.Append(fld);
+    }
+    cbsd.Set("lockFields", std::move(lfs));
+  }
+
+  if (session_data->redirect_url) {
+    cbsd.Set("redirectUrl", session_data->redirect_url.value());
+  }
+
+  cbsd.Set("serviceProvider", session_data->service_provider);
+
+  if (session_data->source_amount) {
+    cbsd.Set("sourceAmount", session_data->source_amount.value());
+  }
+
+  base::Value::List source_currency_codes;
+  for (const auto& cc : session_data->source_currency_codes) {
+    source_currency_codes.Append(cc);
+  }
+  cbsd.Set("sourceCurrencyCodes", std::move(source_currency_codes));
+
+  if (session_data->wallet_address) {
+    cbsd.Set("walletAddress", session_data->wallet_address.value());
+  }
+
+  if (session_data->wallet_tag) {
+    cbsd.Set("walletTag", session_data->wallet_tag.value());
+  }
+
+  cbwr.Set("sessionData", std::move(cbsd));
+  cbwr.Set("sessionType", "TRANSFER");
+
+  FillCustomerData(customer_data, cbwr);
+
+  return cbwr;
+}
+
+std::string GetPayload(const std::optional<base::Value::Dict>& payload_value) {
+  if (!payload_value) {
+    return "";
+  }
+
+  std::string payload;
+  auto serialize_success =
+      base::JSONWriter::Write(payload_value.value(), &payload);
+  DCHECK(serialize_success);
```

Comment:
Is this intentionally set as a DCHECK or did you intend on this executing in a release build? It wasn't clear to me which way it should be done in code so just wanted to double check.

Reply:
No its just debug check

Reply:
Thanks for confirming